src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAGIC_NUMBER = 10;
+    private final Random random = new Random();
```

Comment:
It's better to make the `ColorSupplier` an instance variable instead of creating a new one in the `getRandomFigure` method. This way, you only create one instance of `ColorSupplier` for each `FigureSupplier` instance.